src/sparse/KokkosSparse_sptrsv_superlu.hpp

```diff
@@ -0,0 +1,719 @@
+  std::cout << "    * Total nnz   = " << hr (n) << std::endl;
+  std::cout << "    * nnz / n     = " << hr (n)/n << std::endl;
+  #endif
+
+  // deepcopy
+  Kokkos::deep_copy (values_view, hv);
+  // create crs
+  crsmat_t crsmat("CrsMatrix", n, values_view, static_graph);
+  return crsmat;
+}
+
+/* ========================================================================================= */
+// store numerical values of SuperLU U-factor into CSC
+template <typename crsmat_t, typename graph_t>
+crsmat_t read_superlu_valuesU_CSC(bool invert_diag, bool invert_offdiag,
+                                  SuperMatrix *L,  SuperMatrix *U, graph_t &static_graph) {
+
+  using values_view_t  = typename crsmat_t::values_type::non_const_type;
+  using       scalar_t = typename values_view_t::value_type;
+  using integer_view_host_t = Kokkos::View<int*, Kokkos::HostSpace>;
+  using STS = Kokkos::Details::ArithTraits<scalar_t>;
+
+  scalar_t one  = STS::one ();
+  scalar_t zero = STS::zero ();
+
+  SCformat *Lstore = (SCformat*)(L->Store);
+  scalar_t *Lx = (scalar_t*)(Lstore->nzval);
+
+  NCformat *Ustore = (NCformat*)(U->Store);
+  scalar_t *Uval = (scalar_t*)(Ustore->nzval);
+
+  int n = L->nrow;
+  int nsuper = 1 + Lstore->nsuper;     // # of supernodal columns
+  int *nb = Lstore->sup_to_col;
+  int *mb = Lstore->rowind_colptr;
+  int *colptrL = Lstore->nzval_colptr;
+  int *colptrU = Ustore->colptr;
+  int *rowindU = Ustore->rowind;
+
+  /* create a map from row id to supernode id */
+  integer_view_host_t map ("map", n);
+  int supid = 0;
+  for (int k = 0; k < nsuper; k++) {
+    int j1 = nb[k];
+    int j2 = nb[k+1];
+    for (int j = j1; j < j2; j++) {
+        map (j) = supid;
+    }
+    supid ++;
+  }
+
+  auto rowmap_view = static_graph.row_map;
+  auto hr = Kokkos::create_mirror_view (rowmap_view);
+  Kokkos::deep_copy (hr, rowmap_view);
+
+  /* Upper-triangular matrix */
+  int nnzA = hr (n);
+  values_view_t values_view ("values_view", nnzA);
+  auto hv = Kokkos::create_mirror_view (values_view);
+  Kokkos::deep_copy (hv, zero); // seems to be needed in complex (instead of zeroing out lower-tri as below)
+
+  integer_view_host_t sup ("sup", nsuper);
+  integer_view_host_t off ("off", nsuper);
+  integer_view_host_t check ("check", nsuper);
+  Kokkos::deep_copy (check, 0);
+  for (int k = 0; k < nsuper; k++) {
+    int j1 = nb[k];
+    int nscol = nb[k+1] - j1;
+
+    int i1 = mb[j1];
+    int nsrow = mb[j1+1] - i1;
+
+    /* the diagonal "dense" block (!! first !!)*/
+    int psx = colptrL[j1];
+    if (invert_diag) {
+      if (std::is_same<scalar_t, double>::value == true) {
+        LAPACKE_dtrtri (LAPACK_COL_MAJOR,
+                        'U', 'N', nscol,
+                        reinterpret_cast <double*> (&Lx[psx]), nsrow);
+      } else {
+        LAPACKE_ztrtri (LAPACK_COL_MAJOR,
+                        'U', 'N', nscol,
+                        reinterpret_cast <lapack_complex_double*> (&Lx[psx]), nsrow);
+      }
```

Comment:
What about `float` and `complex<float>`?  kokkos-kernels is supposed to support at least the four types that the BLAS supports.  Please at least `static_assert` or throw here.